app/models/assignment.rb

```diff
@@ -379,42 +379,125 @@ def clone_groupings_from(assignment_id)
   # Add a group and corresponding grouping as provided in
   # the passed in Array.
   # Format: [ groupname, repo_name, member, member, etc ]
-  # Any member names that do not exist in the database will simply be ignored
-  # (This makes it possible to have empty groups created from a bad csv row)
+  # The groupname, repo_name must not pre exist, each member should exist and
+  # not belong to a different grouping for the same assignment.
+  # If these requirements are not satisfied, the group and the grouping is
+  # not created.
   def add_csv_group(row)
     return if row.length.zero?
 
     row.map! { |item| item.strip }
 
-    # Note: We cannot use find_or_create_by here, because it has its own
-    # save semantics. We need to set and save attributes in a very particular
-    # order, so that everything works the way we want it to.
     group = Group.where(group_name: row.first).first
-    if group.nil?
-      group = Group.new
-      group.group_name = row[0]
+
+    unless group.nil?
+      if group.repo_name != row[1]
+        # CASE: Group already exits but the repo name is different
+        duplicate_group_error = I18n.t('csv.group_with_different_repo',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.